src/os_amiga.c

```diff
@@ -664,9 +664,20 @@ mch_can_restore_icon(void)
     int
 mch_get_user_name(char_u *s, int len)
 {
+#if defined(__amigaos4__) || defined(__AROS__) || defined(__MORPHOS__)
+    int uid = getuid();
+    struct passwd *pwd = getpwuid(uid);
+
+    if (pwd && pwd->pw_name && len > 0)
+    {
+        vim_strncpy(s, (char_u *) pwd->pw_name, len - 1);
+        return OK;
+    }
```

Comment:
The function has no return statement if the condition at line 671 is false (→ undefined behavior).

Reply:
Silly misstake. Fixed it now.